pytorch_pfn_extras/training/extension.py

```diff
@@ -222,3 +224,48 @@ def decorator(ext: 'ExtensionLike') -> 'ExtensionLike':
 
 def _as_extension(ext: 'ExtensionLike') -> Extension:
     return ext if isinstance(ext, Extension) else _WrappedExtension(ext)
+
+
+class ExtensionEntry:
+    """Extension and options.
+
+    Args:
+        extension: An extension.
+        name: Name of extension. If omitted, inferred from the attributes of
+            ``extension`` argument.
+        priority: Invocation priority of the extension.
+            Default is ``PRIORITY_READER``.
+        trigger: Trigger object that determines when to invoke the extension.
+            Default is ``(1, 'iteration')``.
+        call_before_training (bool): Flag to call extension before training.
+            Default is ``False``.
+
+    .. seealso::
+       :meth:`pytorch_pfn_extras.training.ExtensionsManager.extend`
+    """
+
+    def __init__(
+            self,
+            extension: Extension,
+            name: Optional[str] = None,
+            priority: int = Extension.priority,
+            trigger: 'TriggerLike' = Extension.trigger,
+            call_before_training: bool = False,
+    ) -> None:
+        self.name = name or extension.name
+        self.extension = extension
```

Comment:
Please check that given instance is an Extension so that users do not pass callable here.